core/commit_test.go

```diff
@@ -221,160 +236,85 @@ func TestMakeCommitmentCollectorConcurrent(t *testing.T) {
 }
 
-func TestMakeCommitmentCounter(t *testing.T) {
-	// fault tolerance -> list of cases
-	cases := map[int][]struct {
-		desc      string
-		view      int
-		prepareCV int
-		replicaID int
-		ok        bool
-		done      bool
-	}{
-		// f=1
-		1: {{
-			desc:      "Commitment from primary",
-			prepareCV: 1,
-			replicaID: 0,
-			ok:        true,
-			done:      false,
-		}, {
-			desc:      "One commitment from backup replica is enough",
-			prepareCV: 1,
-			replicaID: 1,
-			ok:        true,
-			done:      true,
-		}, {
-			desc:      "Extra commitment from another backup replica",
-			prepareCV: 1,
-			replicaID: 2,
-			ok:        true,
-			done:      true,
-		}, {
-			desc:      "Second commitment from primary",
-			prepareCV: 2,
-			replicaID: 0,
-			ok:        true,
-			done:      false,
-		}, {
-			desc:      "Third commitment from primary",
-			prepareCV: 3,
-			replicaID: 0,
-			ok:        true,
-			done:      false,
-		}, {
-			desc:      "Non-sequential commitment from backup replica",
-			prepareCV: 3,
-			replicaID: 2,
-			ok:        false,
-			done:      false,
-		}, {
-			desc:      "First commitment in a new view",
-			view:      1,
-			prepareCV: 2,
-			replicaID: 1,
-			ok:        true,
-			done:      false,
-		}, {
-			desc:      "Second commitment in a new view",
-			view:      1,
-			prepareCV: 3,
-			replicaID: 1,
-			ok:        true,
-			done:      false,
-		}, {
-			desc:      "Commitment for old view",
-			view:      0,
-			prepareCV: 2,
-			replicaID: 2,
-			ok:        true,
-			done:      false,
-		}, {
-			desc:      "Non-sequential commitment in a new view",
-			view:      1,
-			prepareCV: 3,
-			replicaID: 0,
-			ok:        false,
-		}, {
-			desc:      "First valid commitment from backup in a new view",
-			view:      1,
-			prepareCV: 2,
-			replicaID: 2,
-			ok:        true,
-			done:      true,
-		}},
-
-		// f=2
-		2: {{
-			desc:      "Commitment from primary",
-			prepareCV: 1,
-			replicaID: 0,
-			ok:        true,
-			done:      false,
-		}, {
-			desc:      "First commitment from backup replica",
-			prepareCV: 1,
-			replicaID: 1,
-			ok:        true,
-		}, {
-			desc:      "Another commitment from primary",
-			prepareCV: 2,
-			replicaID: 0,
-			ok:        true,
-			done:      false,
-		}, {
-			desc:      "Another commitment for another Prepare",
-			prepareCV: 2,
-			replicaID: 1,
-			ok:        true,
-		}, {
-			desc:      "Duplicate commitment is not okay",
-			prepareCV: 1,
-			replicaID: 1,
-			ok:        false,
-		}, {
-			desc:      "Another commitment from backup replica is enough",
-			prepareCV: 1,
-			replicaID: 3,
-			ok:        true,
-			done:      true,
-		}, {
-			desc:      "The second Prepared request is done",
-			prepareCV: 2,
-			replicaID: 2,
-			ok:        true,
-			done:      true,
-		}, {
-			desc:      "Extra commitment for the first request",
-			prepareCV: 1,
-			replicaID: 2,
-			ok:        true,
-			done:      true,
-		}},
+func TestMakeCommitmentAcceptor(t *testing.T) {
```

Comment:
It would be good if we have error cases (which test that inputs cause an expected error) but it may be future work.

Reply:
I am not sure if I understood exactly, but the `ok` test case parameter requires that either nil or non-nil error must be returned. Thus the test would fail in case of unexpected error.

Reply:
Sorry, I was wrong. Error cases are tested.